### ISSUE TITLE
Handling of <section id="issue-summary"> is poor

### ISSUE THREAD

**Issue by AmeliaBR** [w3c/respec#2207]
## Important info

* URL to affected spec: (not online yet) 
* ReSpec version: 24.7.0
* [x] I did a "hard refresh", but it's still busted. 

## Description of problem

ReSpec doesn't work well if you try to customize the Issue Summary.

According to [the wiki on `issue-summary`](https://github.com/w3c/respec/wiki/issue-summary),
if you create a section with that `id`, it will be used for the auto-generated issue summary.  Like similar auto-generated sections, I assumed that meant I could add my own heading and intro paragraph.

**What happened?:** 

The auto generated issue summary generates and adds it's own heading even if I've included one, so I end up with two `<h2>` elements in the section.

If I _don't_ include my own heading in the section, I get the "no-headingless-sections" linter error!

**Minimal test case (for the body of a generic ReSpec document):**

```html
<p class="issue">Write the rest of this spec!</p>

<section class="appendix" id="issue-summary">
<h2>Open Issues</h2><!-- or remove this to see the linter error -->
<p>
We are tracking issues on
<a href="https://github.com/w3c/respec/issues">Github</a>.
</p>
</section>
```

**Comment by marcoscaceres** [comment]
Thanks for the test case and details @AmeliaBR. Will try to fix soon.

**Comment by marcoscaceres** [comment]
@pradeepgangwar do you want to work on this one? The bug happens somewhere here: 
https://github.com/w3c/respec/blob/develop/src/core/issues-notes.js#L138

**Comment by pradeepgangwar** [comment]
@marcoscaceres I have figured out something for previous bug you gave (hope to make a WIP PR soon). So yes, I will also try this one. :D

**Comment by marcoscaceres** [comment]
@AmeliaBR, shipped the fix in ReSpec 24.9.1. Let us know if its working for you.
